app/models/concerns/facets/managed_host_extensions.rb

```diff
@@ -0,0 +1,50 @@
+require 'facets'
+
+module Facets
+  module ManagedHostExtensions
+    extend ActiveSupport::Concern
+
+    included do
+      Facets::ManagedHostExtensions.refresh_facet_relations(self)
+      after_save :clear_association_cache #should be removed after moving to rails 4. Fixes an issue with save! that breaks :inverse_of.
+
+      def attributes
+        hash = super
+
+        # include all facet attributes by default
+        host_facets_with_definitions.each do |facet, facet_definition|
+          hash["#{facet_definition.name}_attributes"] = facet.attributes.reject { |key, _| %w(created_at updated_at).include? key }
```

Comment:
Actually I did it in my [puppet implementation](https://github.com/theforeman/foreman/pull/2555/files#diff-1fcf71199e401fa452c2a82f900b6ed6R12), but I don't remember why...

I'll add it to `#register...` method.